jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -600,6 +603,36 @@ public Request getWrapped()
         }
     }
 
+    static Predicate<Request> forMethod(String... included)
+    {
+        return forMethod(List.of(included), Collections.emptyList());
+    }
+
+    static Predicate<Request> forMethod(Collection<String> included, Collection<String> excluded)
+    {
+        IncludeExclude<String> methods = new IncludeExclude<>();
+        if (included != null)
+            methods.include(included.toArray(new String[0]));
+        if (excluded != null)
+            methods.exclude(excluded.toArray(new String[0]));
+        return r -> methods.test(r.getMethod());
+    }
+
+    static Predicate<Request> forPath(String... included)
+    {
+        return forPath(List.of(included), Collections.emptyList());
+    }
+
+    static Predicate<Request> forPath(Collection<String> included, Collection<String> excluded)
+    {
+        IncludeExclude<String> paths = new IncludeExclude<>(PathSpecSet.class);
+        if (included != null)
+            paths.include(included.toArray(new String[0]));
+        if (excluded != null)
+            paths.exclude(excluded.toArray(new String[0]));
+        return r -> paths.test(r.getPathInContext());
+    }
+
```

Comment:
So in XML this could look like:
```XML
  <New class="org.eclipse.jetty.server.handler.Handler$ConditionalBuilder">
    <Call name="forMethod">
      <Arg>GET</Arg>
    </Call>
    <Call name="forPaths">
      <Arg>
        <Array type="String">
          <Item>"/include/*"</Item>
        </Array>
      </Arg>
    </Call>
    <Call id="ConditionalHandler" name="build">
      <Call name="setHandler">
        ...
      </Call>
    </Call>
```
Which reads OK, but cannot itself be used as an Arg. The resulting handler needs to be saved as an ID and then the reference used to set the handler.   So I'm not so convinced that the builder style works in XML.... perhaps we can add a `<Build class="org.eclipse.jetty.server.handler.Handler$Conditional">...</Build>` element to better support it?   Do we have other examples of builders used in XML?

But perhaps just doing constructor injection and getting rid of the setters would be simplest?

Reply:
```xml
<Call id="conditionalHandler" class="Handler.Conditional" name="forMethod">
  <Arg>GET</Arg>
  <Call name="forPaths">
    <Arg>/include/*</Arg>
    <Call name="forHeader">
      <Arg>X-Custom</Arg>
      ...
    </Call>
  </Call>
</Call>
```

The `<Call>` elements are nested rather than sequential.
I don't see any problem to assign an ID and it can be used as an `Arg`, so I think we're already ok.

Reply:
The calls can either be nested or sequential as builder methods just return this.  It should always be the same to do `builder.foo().bar();` or `builder.foo(); builder.bar()`

I'll make the changes to GzipHandler to use a Conditional Handler and see how it looks..... stand by ....

Reply:
I've pondered the builder style for this, and decided against it.    All our other handlers are built by normal dependency injection, not constructor injection, so I don't think we should change style for just this one handler.

More over, our module configuration mechanism is additive, so we frequently will look for an entity as a XmlConfiguration ID and then add configuration to it.  In this case, I can imagine modules adding predicates to an already installed ConditionalHandler.